sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RequestTimeoutException.java

```diff
@@ -76,6 +77,23 @@ public RequestTimeoutException(String message, HttpHeaders headers, URI requestU
                 : null);
     }
 
+    /**
+     * Instantiates a new Request timeout exception.
+     *
+     * @param message the message
+     * @param headers the headers
+     * @param remoteAddress the remote address
+     */
+    public RequestTimeoutException(String message, HttpHeaders headers, SocketAddress remoteAddress) {
+        super(message,
+            null,
+            HttpUtils.asMap(headers),
+            HttpConstants.StatusCodes.REQUEST_TIMEOUT,
+            remoteAddress != null
```

Comment:
Isn't null value a bug? 
Or when is it expected.

Reply:
Not expected - null check here to avoid NPE - and to be consistent with other overloads. So consumers of the RequestTimeoutException have to be aware that it can be null today already.

Leaving it as is - please reactivate if you disagree.